### ISSUE TITLE
Set default test size to "small"

### ISSUE THREAD

**Issue by matts1** [bazelbuild/bazel#22355]
### Description of the feature request:

If unspecified, a test's size currently defaults to "medium". I think small is a far more reasonable default.

[aspect_bazel_lib](https://github.com/aspect-build/bazel-lib/blob/1697a3275becc29d0c002f6ef18bd63885b4d758/lib/private/utils.bzl#L139-L171) says:

```
The [test-encyclopedia](https://bazel.build/reference/test-encyclopedia) says:
* Tests may return arbitrarily fast regardless of timeout.
* A test is not penalized for an overgenerous timeout, although a warning may be issued:
* You should generally set your timeout as tight as you can without incurring any flakiness.

However Bazel's default for timeout is medium, which is dumb given this guidance.
```

I'm inclined to agree with this. The *vast* majority of tests should be marked as small, so it would be more convenient, and given the guidance, it would make sense to set the timeout as tight as is allowed (small) by default, and then bazel will timeout and tell you to increase the timeout.

To achieve this, I request creating a flag `--incompatible_default_test_size` and default it to medium. Then in a future release, we can change it to small, before eventually deleting the flag.

### Which category does this issue belong to?

Java Rules

### What underlying problem are you trying to solve with this feature?

Our pre-upload checks add the flag `--test_size_filters=small`. I just recently ran into an issue where the size of a test was not specified, and so the test was skipped. I think defaulting it to small is far more useful, and also means that users don't have to remember to set the test size - instead, they will know to set the test size because the test fails from timing out.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

7.1.1

**Comment by guille2718** [comment]
This really makes a lot of sense. Are there any opposing views?

**Comment by kalvdans** [comment]
Copying in my comments from the duplicate report:

"There were tests whose specified size is too big" shown by default when adding a new test, following the [tutorial](https://bazel.build/start/go#test-project). We should change the default [size](https://bazel.build/reference/be/common-definitions#common-attributes-tests) parameter to "small" to encourage people to write quick unit tests, and force people to annotate when they write slower tests.

**Comment by guille2718** [comment]
I agree that the right long-term state is "tests should be small by default". I think that's what matts1 is proposing as well.

The flag is just an intermediate step towards that goal, to make the change a little smoother.
